src/passes/OptimizeInstructions.cpp

```diff
@@ -917,6 +917,42 @@ struct OptimizeInstructions
     if (binary->right->is<Const>()) {
       return;
     }
+    // Prefer subexpressions with constants on the right as well.
+    if (auto* left = binary->left->dynCast<Binary>()) {
+      if (left->right->is<Const>()) {
+        bool shouldSwap = true;
+        // don't swap if another expression also contain constant
+        // which equal to -1 or 0
+        if (auto* right = binary->right->dynCast<Binary>()) {
+          if (auto rightRightConst = right->right->dynCast<Const>()) {
+            auto value = rightRightConst->value.getInteger();
```

Comment:
nope because `isSymmetric` allows only [integer operations](https://github.com/WebAssembly/binaryen/blob/0dff178cb4467b07d7bd58713e80b46d15601757/src/ir/properties.h#L38). But I could add extra `isInteger` check

Reply:
yes, it seems this could happened. Will fix